indra_db/readonly_dumping/export_assembly.py

```diff
@@ -0,0 +1,984 @@
+                if not validate_statement_semantics(stmt):
+                    continue
+
+                if sh not in seen_hashes:
+                    writer_uniq.writerow((sh, stmt_json_str))
+                    seen_hashes.add(sh)
+    else:
+        logger.info(
+            f"Unique statements already dumped at {unique_stmts_fpath.as_posix()}, "
+            f"skipping..."
+        )
+
+
+def load_statements_from_file(file_path):
+    stmts = []
+    with gzip.open(file_path, "rt") as f:
+        reader = csv.reader(f, delimiter="\t")
+        for _, sjs in reader:
+            stmt = stmt_from_json(clean_json_loads(sjs, remove_evidence=True))
+            stmts.append(stmt)
+    return stmts
+
+def calculate_belief(
+        refinements_graph: nx.DiGraph,
+        num_batches: int,
+        batch_size: int,
+        unique_stmts_path: Path = unique_stmts_fpath,
+):
+    # The refinement set is a set of pairs of hashes, with the *first hash
+    # being more specific than the second hash*, i.e. the evidence for the
+    # first should be included in the evidence for the second
+    #
+    # The BeliefEngine expects the refinement graph to be a directed graph,
+    # with edges pointing from *more specific* to *less specific* statements
+    # (see docstring of indra.belief.BeliefEngine)
+    #
+    # => The edges represented by the refinement set are the *same* as the
+    # edges expected by the BeliefEngine.
+
+    # Initialize a belief engine
+    logger.info("Initializing belief engine")
+    be = BeliefEngine(refinements_graph=refinements_graph)
+
+    # Load the source counts
+    logger.info("Loading source counts")
+    with source_counts_fpath.open("rb") as fh:
+        source_counts = pickle.load(fh)
+
+    # Store hash: belief score
+    belief_scores = {}
+
+    def _get_support_evidence_for_stmt(stmt_hash: int) -> List[Evidence]:
+        # Find all the statements that refine the current
+        # statement, i.e. all the statements that are more
+        # specific than the current statement => look for ancestors
+        # then add up all the source counts for the statement
+        # itself and the statements that refine it
+        summed_source_counts = Counter(source_counts[stmt_hash])
+
+        # If there are refinements, add them to the source counts
+        if stmt_hash in refinements_graph.nodes():
+            refiner_hashes = nx.ancestors(G=refinements_graph,
+                                          source=stmt_hash)
+            for refiner_hash in refiner_hashes:
+                summed_source_counts += Counter(source_counts[refiner_hash])
+
+        # Mock evidence - todo: add annotations?
+        # Add evidence objects for each source's count and each source
+        ev_list = []
+        for source, count in summed_source_counts.items():
+            for _ in range(count):
+                ev_list.append(Evidence(source_api=source))
```

Comment:
This needs to updated with mapping from names used in the source counts to the source api name since `source != source_api` for certain knowledgebases. I'll take care of that.

Reply:
Basically, we rename the variable "source" with "source_api"? Since the key in source_count is source_api and the logic does not change.

Reply:
Updates made in 384c3918156374a1ec7546d4c8c2b9815bc9d253